service/worker/esanalyzer/workflow.go

```diff
@@ -166,6 +174,135 @@ func (w *Workflow) workflowFunc(ctx workflow.Context) error {
 			return err
 		}
 	}
+
+	err = workflow.ExecuteActivity(
+		workflow.WithActivityOptions(ctx, findLongRunningWorkflowsOptions),
+		findLongRunningWorkflowsActivity,
+	).Get(ctx, nil)
+
+	return err
+}
+
+func getLongRunningWorkflowsQuery(
+	maxWorkflowStartTime int64,
+	domainID string,
+	workflowType string,
+) (string, error) {
+	wfTypeMarshaled, err := json.Marshal(workflowType)
+	if err != nil {
+		return "", err
+	}
+	// No need to marshal domainID: it comes from domainEntry and its type is uuid
+	return fmt.Sprintf(`
+    {
+      "query": {
+          "bool": {
+              "must": [
+                  {
+                      "range" : {
+                          "StartTime" : {
+                              "lte" : "%d"
+                          }
+                      }
+                  },
+                  {
+                      "match" : {
+                          "DomainID" : "%s"
+                      }
+                  },
+                  {
+                      "match" : {
+                          "WorkflowType" : %s
+                      }
+                  }
+              ],
+              "must_not": {
+                "exists": {
+                  "field": "CloseTime"
+                }
+              }
+          }
+      },
+      "size": 0
+    }
+    `, maxWorkflowStartTime, domainID, string(wfTypeMarshaled)), nil
+}
+
+// findLongRunningWorkflows is activity to find long running workflows
+func (w *Workflow) findLongRunningWorkflows(ctx context.Context) error {
+	logger := activity.GetLogger(ctx)
+
+	workflowThresholds := w.analyzer.config.ESAnalyzerWorkflowDurationWarnThresholds()
+	if len(workflowThresholds) == 0 {
+		logger.Info("No workflow execution time thresholds defined")
+		return nil
+	}
+
+	var entries map[string]string
+	err := json.Unmarshal([]byte(workflowThresholds), &entries)
+	if err != nil {
+		return err
+	}
+
+	domainNameToID := map[string]string{}
+	for domainWFTypePair, threshold := range entries {
+		index := strings.Index(domainWFTypePair, "/")
+		// -1 no delimiter, 0 means the entry starts with /
+		if index < 1 || len(domainWFTypePair) <= (index+1) {
+			return types.InternalServiceError{
+				Message: fmt.Sprintf("Bad Workflow type entry '%v'", domainWFTypePair),
```

Comment:
for quoted strings, you can use %q 
```suggestion
				Message: fmt.Sprintf("Bad Workflow type entry '%v'", domainWFTypePair),
```